openapi/platform/platform.yaml

```diff
@@ -3665,6 +3916,137 @@ components:
+                fieldType: string
+        Custom Fields with integer value Example:
+          data:
+            id: U-12345678-abcd-1234-abcd-123456789abd
+            type: userCustomFields
+            partnerFields:
+              - fieldID: FieldID-123448ca-abc0-abc5-abc2-abc640bf4b99
+                integerValue: 1000
+                title: Title of 123448ca field
+                description: Description for 123448ca field
+                fieldType: integer
+        Custom Fields with Date value Example:
+          data:
+            id: U-12345678-abcd-1234-abcd-123456789abe
+            type: userCustomFields
+            partnerFields:
+              - fieldID: FieldID-987448ca-abc0-abc5-abc2-abc640bf4b99
+                dateValue: '2016-01-02T15:04:05Z'
+                title: Title of 987448ca field
+                description: Description for 987448ca field
+                fieldType: date
+        Custom Fields Full:
+          data:
+            id: U-12345678-abcd-1234-abcd-123456789abf
+            type: userCustomFields
+            partnerFields:
+              - fieldID: FieldID-abc448ca-abc0-abc5-abc2-abc640bf4b99
+                stringValue: A good string value
+                title: Title of abc448ca field
+                description: Description for abc448ca field
+                fieldType: string
+              - fieldID: FieldID-123448ca-abc0-abc5-abc2-abc640bf4b99
+                integerValue: 1000
+                title: Title of 123448ca field
+                description: Description for 123448ca field
+                fieldType: integer
+              - fieldID: FieldID-987448ca-abc0-abc5-abc2-abc640bf4b99
+                dateValue: '2022-01-02T15:04:05Z'
+                title: Title of 987448ca field
+                description: Description for 987448ca field
+                fieldType: date
+      x-tags:
+        - User Custom Fields
+      properties:
+        id:
+          type: string
+          example: U-12345678-abcd-1234-abcd-123456789abc
+          description: 'An identifier for the parent entity, in this case user'
+          readOnly: true
+        type:
+          type: string
+          default: userCustomFields
+          enum:
+            - userCustomFields
+          description: Should always be userCustomFields
+        partnerFields:
+          type: array
+          description: |-
+            List of custom fields specific to the partner with a direct relationship to the user/contact.
+            To send a new value each field must have at least the fieldId and its correspondent value.
+            For example a field of type string must have the stringValue attribute filled.
+
+            string -> stringValue
+
+            integer -> integerValue
+
+            date -> dateValue
+
+            Only the fields that you wish to modify need to be sent.
+          items:
+            type: object
+            properties:
+              fieldID:
```

Comment:
I just noticed you followed the golang convention of using an uppercase D instead of Google's [JSON convention which we are following](https://vendasta.jira.com/wiki/spaces/API/pages/819626067/Endpoint+Design+Considerations#:~:text=Follow%20this%20formula%20if%20unsure.%20Only%20the%20first%20letter%20of%20an%20acronym%20or%20abbreviation%20should%20be%20upper%20case.%20(somethingId%2C%20somethingIos%2C%20somethingIpv4)).

This is going to need to be fixed for the sales accounts as well in the docs and on the server. I checked [the logs](https://cloudlogging.app.goo.gl/MWHgP3BuSuRJiZJ67) and it does not look like anyone is using the endpoints yet so you should be safe to make the breaking change.

Reply:
I'm gonna throw a PR for that right now

Reply:
But I think it's just for documentation and yaml. For example on the [json tag](https://github.com/vendasta/api-gateway/blob/f003bbd59b09adedf45b0a0ac2be674185d65d4c/internal/storage/customfields/customField.go#L34) it's correct

Reply:
And checking the actual service, it's accepting `fieldId` as name:
<img width="773" alt="Screen Shot 2022-01-26 at 1 52 34 PM" src="https://user-images.githubusercontent.com/73379/151236682-ee47c483-aa7f-440b-a994-5cb761c7e206.png">

Reply:
Hmm. Looks like you are right. We should update the integration test runner to verify that the response only includes fields that are documented.